lib/email_messages.py

```diff
@@ -64,24 +64,27 @@ def email_course_error(request, exercise, message, exception=True):
         pass
 
 
-def email_course_students(
+def email_course_students_and_staff(
         instance: CourseInstance,
         subject: str,
         message: str,
-        audience: Enum = CourseInstance.ENROLLMENT_AUDIENCE.ALL_USERS,
+        student_audience: Enum,
+        include_staff: bool,
         ) -> int:
     """
-    Sends an email to students on the course. Audience parameter controls whether the mail goes
-    to all (default), just internal, or just external students.
+    Sends an email to students and staff of the course. Audience parameter controls whether the mail goes
+    to all, just internal, or just external students.
     Returns number of emails sent, or -1 in case of error.
     """
-    students = instance.students
-    if audience == CourseInstance.ENROLLMENT_AUDIENCE.INTERNAL_USERS:
-        students = students.filter(organization=settings.LOCAL_ORGANIZATION)
-    elif audience == CourseInstance.ENROLLMENT_AUDIENCE.EXTERNAL_USERS:
-        students = students.exclude(organization=settings.LOCAL_ORGANIZATION)
+    student_querys = {
+        CourseInstance.ENROLLMENT_AUDIENCE.ALL_USERS: instance.students,
+        CourseInstance.ENROLLMENT_AUDIENCE.INTERNAL_USERS: instance.students.filter(organization=settings.LOCAL_ORGANIZATION),
+        CourseInstance.ENROLLMENT_AUDIENCE.EXTERNAL_USERS: instance.students.exclude(organization=settings.LOCAL_ORGANIZATION),
+    }
+    recipients = student_querys.get(student_audience, UserProfile.objects.none()).values_list("user__email", flat=True)
+    if include_staff:
+        recipients = recipients.union(instance.course_staff.values_list("user__email", flat=True))
 
-    recipients = students.exclude(user__email='').values_list("user__email", flat=True)
```

Comment:
The new code is missing the part for excluding empty emails. It is best to keep it